lib/spack/spack/traverse.py

```diff
@@ -400,6 +400,8 @@ def traverse_topo_edges_generator(edges, visitor, key=id, root=True, all_edges=F
     # maps parent identifier to a list of edges, where None is a special identifier
     # for the artificial root/source.
     node_to_edges = defaultdict(list)
+    # wrap supplied visitor to track visited edges
+    visitor = CoverEdgesVisitor(visitor, key=key, visited=None)
```

Comment:
In the common case, visitor is already a CoverEdgesVisitor.

Can you apply

```diff
diff --git a/lib/spack/spack/traverse.py b/lib/spack/spack/traverse.py
index cd9faef147..57d09028aa 100644
--- a/lib/spack/spack/traverse.py
+++ b/lib/spack/spack/traverse.py
@@ -482,7 +482,7 @@ def traverse_edges(
 
     # In topo traversal we need to construct a sub-DAG including all unique edges even if we are
     # yielding a subset of them, hence "edges".
-    _cover = "edges" if order == "topo" else cover
+    _cover = "paths" if order == "topo" else cover
     visitor = get_visitor_from_args(_cover, direction, deptype, key, visited)
     root_edges = with_artificial_edges(specs)
```

to avoid double bookkeeping?